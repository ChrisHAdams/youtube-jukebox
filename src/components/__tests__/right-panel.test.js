import React from "react"
import renderer from "react-test-renderer"
import {render, fireEvent, cleanup} from 'react-testing-library'
import RightPanel from "../right-panel";
import {toHaveStyle} from 'jest-dom';

expect.extend({toHaveStyle})


describe("RightPanel", () => {

  it("renders correctly", () => {

    const VideoQueue = [];
    const LightTheme = false;

    const tree = renderer
      .create(<RightPanel
                videoQueue={VideoQueue}
                lightTheme={LightTheme}
                moveVideoUpQueue={"callback"}
                moveVideoDownQueue={"callback"}
                removeFromQueue={"callback"}
                playNow={"callback"}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly based on dark theme", () => {

    const VideoQueue = [];
    const LightTheme = false;

    const {getByTestId} = render(<RightPanel
                videoQueue={VideoQueue}
                lightTheme={LightTheme}
                moveVideoUpQueue={"callback"}
                moveVideoDownQueue={"callback"}
                removeFromQueue={"callback"}
                playNow={"callback"}/>);

    expect(getByTestId('rightPanel')).toHaveStyle(`background-color: '#293447'`);
  })

  it("renders correctly based on light theme", () => {

    const VideoQueue = [];
    const LightTheme = true;

    const {getByTestId} = render(<RightPanel
                videoQueue={VideoQueue}
                lightTheme={LightTheme}
                moveVideoUpQueue={"callback"}
                moveVideoDownQueue={"callback"}
                removeFromQueue={"callback"}
                playNow={"callback"}/>);

    expect(getByTestId('rightPanel')).toHaveStyle(`background-color: '#f4f5f7'`);
  })

})


