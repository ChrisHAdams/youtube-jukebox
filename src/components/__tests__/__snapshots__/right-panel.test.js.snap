// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightPanel renders correctly 1`] = `
<div
  className="sc-dnqmqq cNtSNl"
  data-testid="rightPanel"
>
  <div
    className="sc-gzVnrw iobIbK"
  />
</div>
`;
